scrunch/helpers.py

```diff
@@ -70,7 +76,50 @@ def download_file(url, filename):
     return filename
 
 
-def case_expr(rules, name, alias):
+def validate_categories(categories):
+    """
+    Categories can be defined as simple as:
+        categories=[
+            {'id': 1, 'name': 'Yes', 'selected': True},
+            {'id': 2, 'name': 'No'},
+            {'id': 3, 'name': 'Maybe'},
+            {'id': 4, 'name': 'Missing', 'missing': True}
+        ]
+    This method takes care of validating that only 1 is selected, that
+    only one or None is missing and to fill the definition above to
+    match the API requirement:
+        categories=[
+            {'id': 1, 'name': 'Yes', 'missing': False, 'numeric_value': None, 'selected': True},
+            {'id': 2, 'name': 'No', 'missing': False, 'numeric_value': None, 'selected': False},
+            {'id': 3, 'name': 'Maybe', 'missing': False, 'numeric_value': None, 'selected': False},
+            {'id': 4, 'name': 'Missing', 'missing': True, 'numeric_value': None, 'selected': False}
+        ]
+    """
+    defaults = {'missing': False, 'numeric_value': None, 'selected': False}
+    selected_count = 0
+    missing_count = 0
+    for category in categories:
+        if category.get('selected'):
+            selected_count += 1
+        if category.get('missing'):
+            missing_count += 1
+        if not category.get('id'):
```

Comment:
A few extra validations here:
 * ID must be an int
 * IDs must be unique

Similar for names:
  * Must be non empty strings
  * Must be unique
It's also OK if you don't do them though, the server will complain and won't let you anyway.

Reply:
This is cool, but should apply to all other methods that deal with variables, so it's perhaps a good idea to have a ticket to enforce this globally